packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -113,6 +113,7 @@ export class GatewayRoute extends cdk.Resource implements IGatewayRoute {
     const gatewayRoute = new CfnGatewayRoute(this, 'Resource', {
       gatewayRouteName: this.physicalName,
       meshName: props.virtualGateway.mesh.meshName,
+      meshOwner: this.virtualGateway.mesh.env.account,
```

Comment:
Like I explained in https://github.com/aws/aws-cdk/pull/15353#discussion_r660892324, this property should not be present in the resulting template if:

1. Both `env.account` properties of this resource, and `mesh`, are Tokens (`Token.isUnresolved()` returns `true` for them).
2. Both `env.account` properties of this resource, and `mesh`, are _not_ Tokens, and are identical.

You can use the `Token.compareStrings()` API to check this. It will probably require introducing a helper functions shared between all resources that have the `meshOwner` property.

Please also add unit tests that utilize the `ABSENT` helper that verify this behavior.